### PR TITLE
tested filterArray file using jest

### PR DESCRIPTION
**Pull request by snaanortey**
Please review my first jest test. Here is a summary of my test:

I created a test using `jest` for the `filterArray` file. The jest test checks that the `filterArray` function returns an array that does not contain elements from `itemsToExclude`.

### CODE REVIEW

File: server/tsconfig.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
         "noUnusedLocals": true,
         "sourceMap": true,
         "resolveJsonModule": true,
+        "allowJs": true
```
Comment (ehimah) [inline server/tsconfig.json:13]: let's remove this as it wasn't the answer to our problems

File: server/src/services/filterArray.ts [hunk 2]
```diff
@@ -1,9 +1,11 @@
-function filterArray(array: string[], values: string[]): string[] {
+export const itemsToExclude = ["Container", "Food", "Vegetable", "Tableware", "Table", "Spoon", "Fork", "Knife", "Plate"];
+
```
Comment (ehimah) [inline server/src/services/filterArray.ts:2]: why do we have this here? shouldn't this be in your test case?

File: server/src/routes/imageSearch.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ import express from "express";
 import { ImageAnnotatorClient } from "@google-cloud/vision";
 import { RequestHandler } from "express-serve-static-core";
 import fileUpload from "express-fileupload";
+import { filterArray, itemsToExclude } from "../services/filterArray";
```
Comment (ehimah) [inline server/src/routes/imageSearch.ts:7]: okay wait, if the list of items to be filtered is only used in this file, then it should be defined here instead. the filter array is supposed to be a utility function therefore the list should not be defined alongside it

File: server/jest.config.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+/** @type {import('ts-jest').JestConfigWithTsJest} */
```
Comment (ehimah) [inline server/jest.config.js:1]: clean up

File: server/jest.config.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+module.exports = {
+  preset: 'ts-jest',
+  testEnvironment: 'node',
+  moduleFileExtensions: ['ts', 'js', 'json'],
+  transform: {
+    '^.+\\.(ts|js)$': ['ts-jest', {
+      // tsconfig: '<rootDir>/tsconfig.json'
```
Comment (ehimah) [inline server/jest.config.js:8]: clean up

File: server/src/routes/imageSearch.ts [hunk 6]
```diff
@@ -39,8 +40,10 @@ const routeHandler: RequestHandler = async (req, res) => {
       return false;
     });
     const namesArray = annotationsWithNameProperty.map((item) => item.name!);
-    // const response = await searchIngredientsByKeyWords(namesArray);
-    res.send(namesArray);
+
+    const filteredArray = filterArray(namesArray, itemsToExclude);
+
+    res.send(filteredArray);
```
Comment (ehimah) [inline server/src/routes/imageSearch.ts:46]: so basically you need to move the array definition to this file, so you don't have to import it.

File: server/src/services/filterArray.test.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  describe("filterArray", () => {
+    const array = [
+      "Tomato",
+      "Pepper",
+      "Plate",
+      "Cabbage",
+      "Vegetable",
+      "Onion",
+    ];
+    const expected = ["Tomato", "Pepper", "Cabbage", "Onion"];
```
Comment (ehimah) [inline server/src/services/filterArray.test.ts:15]: this test case should have its own `itemsToExclude` array which is different from what you will be using for your application logic.

**Review by ehimah — CHANGES_REQUESTED** [review]

File: server/src/services/filterArray.ts (hunk 2 above)
Comment (snaanortey) [reply server/src/services/filterArray.ts:2]: I have moved this to the specific route file

File: server/tsconfig.json (hunk 1 above)
Comment (snaanortey) [reply server/tsconfig.json:13]: I have deleted this

File: server/src/routes/imageSearch.ts (hunk 3 above)
Comment (snaanortey) [reply server/src/routes/imageSearch.ts:7]: Updated, thanks

File: server/jest.config.js (hunk 4 above)
Comment (snaanortey) [reply server/jest.config.js:1]: I deleted it, and formatted the file.

File: server/jest.config.js (hunk 5 above)
Comment (snaanortey) [reply server/jest.config.js:8]: I deleted the comment

File: server/src/routes/imageSearch.ts (hunk 6 above)
Comment (snaanortey) [reply server/src/routes/imageSearch.ts:46]: I have done that

File: server/src/services/filterArray.test.ts (hunk 7 above)
Comment (snaanortey) [reply server/src/services/filterArray.test.ts:15]: I have done that

**Comment by snaanortey** [comment]
Comments have been addressed and a new commit has been made. Please approve the merge request.

**Comment by snaanortey** [comment]
All comments have been addressed, including updating the array on `itemsToExclude` to include only elements necessary for testing.

**Review by ehimah — APPROVED** [review]
LGTM

**Pull request merged**